### PR TITLE
Mount path for mongo is now configurable

### PR DESCRIPTION
**Pull request by whenhellfreezes**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

Allows for mongo's storage to be mounted at a configurable location. Fixing issue https://github.com/DecipherNow/helm-charts/issues/482

<br/><br/>

2. What **changes to custom.yaml** are required?

None

<br/><br/>

3. Have you documented any additional setup steps or configurations options?

NA

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

tested in minikube.

<br/><br/>

### CODE REVIEW

**Comment by chrisbsmith** [comment]
Nick, please provide a little more information and logs on your testing for this PR. Specifically I’d like to see that you deleted a mongo pod and the data persisted. Maybe even through a deletion of the sts.

**Comment by whenhellfreezes** [comment]
> Nick, please provide a little more information and logs on your testing for this PR. Specifically I’d like to see that you deleted a mongo pod and the data persisted. Maybe even through a deletion of the sts.

I did `describe pod data-mongo-0` with two values. Did not test pod deletion.

**Comment by whenhellfreezes** [comment]
Data persists across pod delete and sts delete however I'm not sure mongo interacts well with being interrupted.

`WiredTiger                            collection-2-6039880098720681677.wt  index-3-6039880098720681677.wt  index-9-6039880098720681677.wt
WiredTiger.lock                       collection-4-6039880098720681677.wt  index-3-6455926780552373423.wt  index-9-6455926780552373423.wt
WiredTiger.turtle                     collection-7-6039880098720681677.wt  index-4-6455926780552373423.wt  journal
WiredTiger.wt                         collection-7-6455926780552373423.wt  index-5-6039880098720681677.wt  mongod.lock
WiredTigerLAS.wt                      diagnostic.data                      index-5-6455926780552373423.wt  sizeStorer.wt
_mdb_catalog.wt                       index-1-6039880098720681677.wt       index-6-6039880098720681677.wt  storage.bson
collection-0-6039880098720681677.wt   index-1-6455926780552373423.wt       index-6-6455926780552373423.wt
collection-0-6455926780552373423.wt   index-11-6455926780552373423.wt      index-8-6039880098720681677.wt
collection-10-6455926780552373423.wt  index-2-6455926780552373423.wt       index-8-6455926780552373423.wt`

Seems like it created a second set of index-{num}-{stamp}.wt files. Not sure if this is expected behaviour.

**Comment by whenhellfreezes** [comment]
It is expected behaviour. Viewed data through mongo cli.

File: data/values.yaml [hunk 1]
```diff
@@ -201,6 +201,7 @@ mongo:
   imagePullPolicy: IfNotPresent
   #private_image: false
   storage:
+    mount_path: /data/db
```
Comment (chrisbsmith) [inline data/values.yaml]: While we have a large problem of mixed base in this file (camel vs. snake), we want to use camel case and we have an open issue ([236](https://github.com/DecipherNow/helm-charts/issues/236)) to update the file to camel case.

To keep this as one less line we need to update in the future, please update your variable to camel case.  And please add a comment that describes the variable's use (another thing we don't do well)

**Pull request merged**